src/index.ts

```diff
@@ -0,0 +1,2721 @@
+  const len = places + 1;
+  const numDigits = Math.ceil(Math.log10(Math.abs(value)));
+  const rounded = Math.round(value * Math.pow(10, len - numDigits)) * Math.pow(10, numDigits - len);
+  return parseFloat(rounded.toFixed(Math.max(len - numDigits, 0)));
+};
+
+const f_maglog10 = function (n: number) {
+  return Math.sign(n) * Math.log10(Math.abs(n));
+};
+
+//from HyperCalc source code
+const f_gamma = function (n: number) {
+  if (!isFinite(n)) {
+    return n;
+  }
+  if (n < -50) {
+    if (n === Math.trunc(n)) {
+      return Number.NEGATIVE_INFINITY;
+    }
+    return 0;
+  }
+
+  let scal1 = 1;
+  while (n < 10) {
+    scal1 = scal1 * n;
+    ++n;
+  }
+
+  n -= 1;
+  let l = 0.9189385332046727; //0.5*Math.log(2*Math.PI)
+  l = l + (n + 0.5) * Math.log(n);
+  l = l - n;
+  const n2 = n * n;
+  let np = n;
+  l = l + 1 / (12 * np);
+  np = np * n2;
+  l = l + 1 / (360 * np);
+  np = np * n2;
+  l = l + 1 / (1260 * np);
+  np = np * n2;
+  l = l + 1 / (1680 * np);
+  np = np * n2;
+  l = l + 1 / (1188 * np);
+  np = np * n2;
+  l = l + 691 / (360360 * np);
+  np = np * n2;
+  l = l + 7 / (1092 * np);
+  np = np * n2;
+  l = l + 3617 / (122400 * np);
+
+  return Math.exp(l) / scal1;
+};
+
+const _twopi = 6.2831853071795864769252842; // 2*pi
+const _EXPN1 = 0.36787944117144232159553; // exp(-1)
+const OMEGA = 0.56714329040978387299997; // W(1, 0)
+//from https://math.stackexchange.com/a/465183
+// The evaluation can become inaccurate very close to the branch point
+const f_lambertw = function (z: number, tol = 1e-10): number {
+  let w;
+  let wn;
+
+  if (!Number.isFinite(z)) {
+    return z;
+  }
+  if (z === 0) {
+    return z;
+  }
+  if (z === 1) {
+    return OMEGA;
+  }
+
+  if (z < 10) {
+    w = 0;
+  } else {
+    w = Math.log(z) - Math.log(Math.log(z));
+  }
+
+  for (let i = 0; i < 100; ++i) {
+    wn = (z * Math.exp(-w) + w * w) / (w + 1);
+    if (Math.abs(wn - w) < tol * Math.abs(wn)) {
+      return wn;
+    } else {
+      w = wn;
+    }
+  }
+
+  throw Error(`Iteration failed to converge: ${z.toString()}`);
+  //return Number.NaN;
+};
+
+//from https://github.com/scipy/scipy/blob/8dba340293fe20e62e173bdf2c10ae208286692f/scipy/special/lambertw.pxd
+// The evaluation can become inaccurate very close to the branch point
+// at ``-1/e``. In some corner cases, `lambertw` might currently
+// fail to converge, or can end up on the wrong branch.
+function d_lambertw(z: Decimal, tol = 1e-10): Decimal {
+  let w;
+  let ew, wew, wewz, wn;
+
+  if (!Number.isFinite(z.mag)) {
+    // FIXME: It checks z.mag, which means it can't be anything but a decimal
+    return z;
+  }
+  // FIXME: Seems like this should be == Decimal.dZero?
+  if (z === 0) {
+    return z;
+  }
+  // FIXME: Seems like this should be == Decimal.dOne?
+  if (z === 1) {
+    //Split out this case because the asymptotic series blows up
+    return OMEGA; // FIXME: Seems like this should return D(OMEGA)?
```

Comment:
```suggestion
    return D(OMEGA);
```
Or possibly a new constant `Decimal.dOmega`?